nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java

```diff
@@ -112,11 +106,71 @@ public void testMetadataProcessing() throws Exception
     verifyTextGzipContent(is(equalTo(agricolae121Content)), contentAfterDelete);
   }
 
-  private void uploadPackage(String... names) throws IOException {
+  @Test
+  public void testDeleteAssetWithComponent() {
+    final Asset asset = findAsset(repository, AGRICOLAE_PATH_FULL_121_TARGZ);
+    assertNotNull(asset);
+    assertNotNull(asset.componentId());
+
+    final Component component = findComponentById(repository, asset.componentId());
+    assertNotNull(component);
+    assertEquals(1, findAssetsByComponent(repository, component).size());
+
+    ComponentMaintenance maintenanceFacet = repository.facet(ComponentMaintenance.class);
+    maintenanceFacet.deleteAsset(asset.getEntityMetadata().getId(), true);
+
+    assertNull(findAsset(repository, AGRICOLAE_PATH_FULL_121_TARGZ));
+    assertNull(findComponentById(repository, asset.componentId()));
+  }
+
+  @Test
+  public void testDeleteAssetWithoutComponent() throws IOException {
```

Comment:
same as above `testDeletingAssetWhenMultipleExistDoesNotDeleteComponent`

Reply:
Done in 6ef47d8db038b9f39d05981d3de9c28952390dbb